serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java

```diff
@@ -175,12 +175,12 @@ Entry getEntry(int i) throws IOException {
             if (buffer.size() > i) {
                 return buffer.get(i);
             } else {
-                if (buffer.size() != i) {
```

Comment:
why is this change necessary? is something calling decodeKey again after it already returned null?

Reply:
Reverted